src/rust/ide/src/controller/searcher.rs

```diff
@@ -209,6 +209,26 @@ impl ParsedInput {
             CompletedFragmentId::Argument {..}                  => TypingArgument,
         }
     }
+
+    /// Convert the current input to Prefix Chain representation.
+    pub fn as_prefix_chain
+    (&self, parser:&Parser) -> Option<ast::Shifted<ast::prefix::Chain>> {
+        let parsed_pattern = parser.parse_line(&self.pattern).ok();
+        let pattern_sast   = parsed_pattern.map(|p| ast::Shifted::new(self.pattern_offset,p));
+        if let Some(mut chain) = self.expression.clone() {
```

Comment:
minor performance thing - instead of cloning here, you can clone after successful match.